### DISCUSSION TITLE
How to do series/parallel animation in RNA2?

### DISCUSSION THREAD

**Discussion by mohsinulhaq** [software-mansion/react-native-reanimated#2011] [Q&A]
I'm looking for the equivalent of `Animated.sequence` and `Animated.parallel`. So far from the docs for v2, I could only see the `withSequence` function that changes the value of only on value and therefore the style of only one component in series.

What I was looking for was to trigger animations in two different components, either in series or in parallel.
For parallel, it seems changing values in statements one after another worked. Correct me if I'm wrong.
```ts
  // these will run in parallel
  val1Shared.value = withTiming(50);
  val2Shared.value = withTiming(100);
```

But for series, I need to have each animation inside a useTiming callback. Which leads to kind of callback hell.

```ts
  val1Shared.value = withTiming(50, undefined, () => {
    val2Shared.value = withTiming(100);
  });
```

Please help with the best practices in achieving this with reanimated 2.
Thanks.

**Comment by svobik7** [comment]
Any update on this? Kinda blocker for me.

**Reply by Noitidart** [reply]
I was also looking for this, did you figure it out? I was hoping to not create one value and then interpolate on that for two different things.

**Comment by MatiPl01** [comment]
Hey!

If someone is still looking for sequential animations, we have the [withSequence](https://docs.swmansion.com/react-native-reanimated/docs/animations/withSequence/) function to achieve this behavior.

I close this discussion as it seems that it refers to previous version of reanimated where `withSequence` wasn't implemented. Fell free to re-open the discussion or create a new one of you have more questions.
